src/kernels/normalkernel.jl

```diff
@@ -12,66 +12,43 @@ eltype(::AbstractNormalKernel{T}) where {T} = T
 
 Standard parametrisation of Normal kernels.
 """
-struct NormalKernel{T,U,V<:AbstractMatrix} <: AbstractNormalKernel{T}
+struct NormalKernel{T,U,V} <: AbstractNormalKernel{T}
     μ::U
     Σ::V
     function NormalKernel(μ, Σ)
         new{eltype(μ),typeof(μ),typeof(Σ)}(μ, Σ)
     end
 end
 
-"""
-    NormalKernel(Φ::AbstractMatrix, Σ::AbstractMatrix)
+const AffineNormalKernel{T} =
+    NormalKernel{T,<:AbstractAffineMap,<:Union{UniformScaling,AbstractMatrix}}
 
-Creates a Normal kernel with linear conditional mean of slope Φ and covariance matrix Σ.
-"""
-NormalKernel(Φ::AbstractMatrix, Σ::AbstractMatrix) = NormalKernel(AffineMap(Φ), Σ)
+const AffineIsoNormalKernel{T} = NormalKernel{T,<:AbstractAffineMap,<:UniformScaling}
 
-"""
-    NormalKernel(Φ::AbstractMatrix, b::AbstractVector, Σ::AbstractMatrix)
+NormalKernel(Φ::AbstractMatrix, Σ) = NormalKernel(AffineMap(Φ), Σ)
 
-Creates a Normal kernel with affine conditional mean of slope Φ, intercept b, and covariance matrix Σ.
-"""
-NormalKernel(Φ::AbstractMatrix, b::AbstractVector, Σ::AbstractMatrix) =
-    NormalKernel(AffineMap(Φ, b), Σ)
-NormalKernel(
-    Φ::AbstractMatrix,
-    b::AbstractVector,
-    pred::AbstractVector,
-    Σ::AbstractMatrix,
-) = NormalKernel(AffineMap(Φ, b, pred), Σ)
+NormalKernel(Φ::AbstractMatrix, b::AbstractVector, Σ) = NormalKernel(AffineMap(Φ, b), Σ)
+NormalKernel(Φ::AbstractMatrix, b::AbstractVector, pred::AbstractVector, Σ) =
+    NormalKernel(AffineMap(Φ, b, pred), Σ)
 
 mean(K::NormalKernel) = K.μ
 
-cov(K::NormalKernel{T,U,V}) where {T,U,V<:AbstractMatrix} = K.Σ
+cov(K::NormalKernel) = K.Σ
 
 condition(K::NormalKernel, x) = Normal(mean(K)(x), cov(K))
 
-compose(
-    K2::NormalKernel{T,U,V},
-    K1::NormalKernel{T,U,V},
-) where {T,U<:AbstractAffineMap,V<:AbstractMatrix} =
+compose(K2::AffineNormalKernel{T}, K1::AffineNormalKernel{T}) where {T} =
```

Comment:
Is the `T` needed, or could this also be:
```suggestion
compose(K2::AffineNormalKernel, K1::AffineNormalKernel) =
```
?

Reply:
Ah is this to make sure that the types interact well?

Reply:
It is needed that both arguments have eltype <: Real or <:Complex. Though maybe this could be solved differently...